src/main/java/org/spongepowered/common/effect/potion/PotionEffectContentUpdater.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.effect.potion;
+
+import com.google.common.collect.ImmutableMap;
+import org.spongepowered.api.CatalogType;
+import org.spongepowered.api.data.DataView;
+import org.spongepowered.api.data.persistence.DataContentUpdater;
+import org.spongepowered.common.data.util.DataQueries;
+import org.spongepowered.common.data.util.DataVersions;
+import org.spongepowered.common.registry.type.effect.PotionEffectTypeRegistryModule;
+
+import java.util.stream.Collectors;
+
+public class PotionEffectContentUpdater implements DataContentUpdater {
+
+    private final static ImmutableMap<String, String> map = ImmutableMap.<String, String>builder().putAll(PotionEffectTypeRegistryModule.getInstance().getAll()
+            .stream().collect(Collectors.toMap(CatalogType::getName, CatalogType::getId))).build();
+
+    @Override
+    public int getInputVersion() {
+        return DataVersions.Potion.BROKEN_POTION_ID;
+    }
+
+    @Override
+    public int getOutputVersion() {
+        return DataVersions.Potion.CURRENT_VERSION;
```

Comment:
If we later create a version 3, won’t this break any really old data being migrated, as the migratory would then mislabel version 2 data as version 3?

This should just be version 2 or its own constant, separate from the `CURRENT_VERSION` constant to avoid this mistake later.

Reply:
Agreed, we can have two pointers, one being the constant `2` for using the proper id, and the "current version" that points to version `2`.